lib/cartopy_mesh_raster/fast_mesh_geometry_calcs.py

```
@@ -108,9 +128,14 @@ def search_faces_for_point(target_point_xyz, i_point_nearest,
                 i.e. n > 1 :  This may occur when close to an edge.
             (0, -1) if not in any of them.
 
+    .. note::
+        the last 3 kwargs are in fact required, and *not* optional.
+        This is just for a nicer arg ordering.
```

How about doing a check that these 3 have been set at the start of the function. To fail early with a nice clear error message?

---

Ok, nice idea !